local-env-setup.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+darwin_install_pip3_packages() {
+    echo "Installing setuptools grpcio-tools virtualenv"
+    pip3 install setuptools grpcio-tools
+    echo "Installing mypy-protobuf"
+    pip3 install --user mypy-protobuf
+}
+
+install_go_packages(){
+        echo "Installing goavro"
+        go get github.com/linkedin/goavro
+        # As we are using bash, we are assuming .bashrc exists.
+        grep -qxF "export GOPATH=${PWD}/sdks/go/examples/.gogradle/project_gopath" ~/.bashrc
+        gopathExists=$?
+        if [ $gopathExists -ne 0 ]; then
+            export GOPATH=${PWD}/sdks/go/examples/.gogradle/project_gopath && echo "GOPATH was set for this session to '$GOPATH'. Make sure to add this path to your ~/.bashrc file after the execution of this script."
+        fi
+}
+
+# Running on Linux
+if [ "$(uname -s)" = "Linux" ]; then
+    # Assuming Debian based Linux and the prerequisites in https://beam.apache.org/contribute/ are met:
+    apt-get update
+
+    echo "Installing dependencies listed in pkglist file"
+    apt-get install -y $(grep -v '^#' dev-support/docker/pkglist | cat) # pulling dependencies from pkglist file
+
+    type -P python3 > /dev/null 2>&1
+    python3Exists=$?
+    type -P pip3 > /dev/null 2>&1
+    pip3Exists=$?
+    if [ $python3Exists -eq 0  -a $pip3Exists -eq 0 ]; then
+        echo "Installing grpcio-tools mypy-protobuf"
+        pip3 install grpcio-tools mypy-protobuf
+    else
+        echo "Python3 and pip3 are required but failed to install. Install them manually and rerun the script."
+        exit
+    fi
+
+    type -P go > /dev/null 2>&1
+    goExists=$?
+    if [ $goExists -eq 0 ]; then
+        install_go_packages
+    else
+        echo "Go is required. Install it manually from https://golang.org/doc/install and rerun the script."
+        exit
+    fi
+fi
+
+# Running on Mac
+if [ "$(uname -s)" = "Darwin" ]; then
```

Comment:
Perhaps this should be structured as:
```
kernelname = $(uname -s)
if kernelname == "Linux"
   ...
elseif kernelname == "Darwin"
  ...
else
  echo "Unrecognized Kernel Name: $kernelname"
```

Right now if we don't hit either if the script just exits silently.